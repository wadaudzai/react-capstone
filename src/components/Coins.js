/* eslint-disable max-len */
/* eslint-disable react-hooks/exhaustive-deps */
import { React, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import OneCoin from './OneCoin';
import '../styles/Coins.css';
import { fetchCoins } from '../redux/coinReducer';

const Coins = () => {
  const coins = useSelector((state) => state.coins);
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');
  const filteredCryptos = coins.filter((coin) => coin.name.toLowerCase().includes(search.toLowerCase()) || coin.symbol.toLowerCase().includes(search.toLowerCase()));
  useEffect(() => {
    dispatch(fetchCoins());
  }, []);
  return (
    <div className="app-holder">
      <div className="form-holder">
        <input type="text" placeholder="Search a cryptocurrency" onChange={(e) => setSearch(e.target.value)} className="form" />
      </div>
      <div className="main-holder">
        {
          filteredCryptos.map((coin) => (
            <OneCoin
              key={coins.data}
              coins={coin}
              search={search}
            />
          ))
        }
      </div>
    </div>
  );
};

export default Coins;
