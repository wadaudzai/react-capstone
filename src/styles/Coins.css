.title {
  color: green;
}

.main-holder {
  display: grid;
  grid-template-columns: 33% 33% 33%;
  width: 100%;
  margin: 0 1rem;
}

.form {
  width: 80%;
  margin: 1rem;
  padding: 10px;
}

.app-holder {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.form-holder {
  background-color: blue;
  width: 100%;
  text-align: center;
}
