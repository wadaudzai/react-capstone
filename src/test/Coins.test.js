import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Coins from '../components/Coins';
import store from '../redux/configureStore';

it('Test Coin', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Coins />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
