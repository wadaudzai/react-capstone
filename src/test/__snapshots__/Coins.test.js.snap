// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Coin 1`] = `
<div
  className="app-holder"
>
  <div
    className="form-holder"
  >
    <input
      className="form"
      onChange={[Function]}
      placeholder="Search a cryptocurrency"
      type="text"
    />
  </div>
  <div
    className="main-holder"
  />
</div>
`;
